### PR TITLE
Fix apply setAutocommit=false only for needed statements

### PR DESCRIPTION
**Pull request by thushaaanthan**
Continue of: https://github.com/wso2/carbon-identity-framework/pull/2217
Issue: wso2/product-is#5538

### CODE REVIEW

File: components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/persistance/ThriftAuthenticationJDBCPersistenceManager.java [hunk 1]
```diff
@@ -103,17 +103,24 @@ private void initDataSource() throws AuthenticationException {
         }
     }
 
+    @Deprecated
```
Comment (ruwanta) [inline components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/persistance/ThriftAuthenticationJDBCPersistenceManager.java]: Need to put a comment why this is deprecated and the alternative to be used.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/JDBCPersistenceManager.java [hunk 2]
```diff
@@ -116,30 +116,38 @@ public void initializeDatabase() {
         dbInitializer.createIdentityDatabase();
     }
 
+    @Deprecated
```
Comment (ruwanta) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/JDBCPersistenceManager.java]: Need a comment why depreciated and the alternative

File: components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/persistance/ThriftAuthenticationJDBCPersistenceManager.java (hunk 1 above)
Comment (thushaaanthan) [reply components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/persistance/ThriftAuthenticationJDBCPersistenceManager.java]: Added

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/JDBCPersistenceManager.java (hunk 2 above)
Comment (thushaaanthan) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/JDBCPersistenceManager.java]: Added

**Pull request merged**